mmcls/models/heads/multi_task_head.py

```diff
@@ -136,26 +114,25 @@ def predict(
             List[MultiTaskDataSample]: A list of data samples which contains
             the predicted results.
         """
-        # The part can be traced by torch.fx
-        predictions_ = dict()
+        predictions_dict = dict()
+
         for task_name, head in self.task_heads.items():
-            predictions_[task_name] = head.predict(feats, data_samples)
-        # The part can not be traced by torch.fx
-        predictions = self._get_predictions(predictions_, data_samples)
-        return predictions
-
-    def _get_predictions(self, preds_dict, data_samples):
-        """Post-process the output of MultiTaskHead."""
-        pred_dicts = [
-            dict(zip(preds_dict, t)) for t in zip(*preds_dict.values())
-        ]
+            if data_samples is None:
+                task_samples = None
+            else:
+                task_samples = [item.get(task_name) for item in data_samples]
+            task_samples = head.predict(feats, task_samples)
```

Comment:
@mzr1996 

here there are cases when the task_samples will be like `[clsDataSample, None, clsDataSample, None, None]`

In this case, it fails in https://github.com/piercus/mmclassification/blob/multi-task/mmcls/models/heads/cls_head.py#L144

In order to keep it generic, i feel we should fix it inside `cls_head`.

Do you agree ?